File: 0189-rotate-array/0189-rotate-array.ts
/**
 Do not return anything, modify nums in-place instead.
 */
function rotate(nums: number[], k: number): void {
    let steps = k%nums.length;
    let copy = [...nums]
    for (let i = 0; i < nums.length; i++){
        nums[(i+steps)%nums.length] = copy[i]
    }
};

File: 0189-rotate-array/0189-rotate-array.java
class Solution {
    public void rotate(int[] nums, int k) {
        int n = nums.length;
         k = k % n;
        if (n == 1) return;
        int[] arr = new int[n];
        int j = 0;
        for (int i = k; i < nums.length; i++){
            arr[i] = nums[j++]; 
        }
        for (int i = 0; i < k; i++){
            arr[i] = nums[j++]; 
        }
        for (int i = 0; i < n; i++){
            nums[i] = arr[i];
        }
    }
}